### PR TITLE
Python 3 compatibility with `chplenv` scripts

### PR DESCRIPTION
**Pull request by ben-albrecht**
* The problem: 
    * `util/chplenv/*py` scripts would fail in Python 3 due to legacy style relative imports, 
         * e.g. `import utils`

* The solution: 
    * Updated to `$PYTHONPATH` to include `$CHPL_HOME/util/chplenv` so absolute imports work in Python 2 & 3 
    * Unfortunately, the 2/3 compatible relative import: `from . import foo`, required a large amount of messy changes throughout the code base, and was ultimately determined to be an inferior solution

* Also tried to clean/sort Python imports with [`isort`](https://github.com/timothycrosley/isort)
    * Still looks a little ugly because the relative imports must follow the `$PYTHONPATH` injection

* Tested with Python 2.7.10 and 3.5.0.

* A new Jenkins job: `correctness-test-python3` will ensure we maintain Python 3 compatibility in the future.

### CODE REVIEW

**Comment by ben-albrecht** [comment]
@Kyle-B - do you want to take a look?

These changes are ready to go (Jenkins job set up and working)

File: util/chplenv/chpl_3p_re2_configs.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 #!/usr/bin/env python
-import utils
+import os
+import sys
 from utils import memoize
```
Comment (Kyle-B) [inline util/chplenv/chpl_3p_re2_configs.py]: Some of these look like they have their imports in the wrong order.

File: util/chplenv/chpl_aux_filesys.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
 #!/usr/bin/env python
+import sys
 import os
-from sys import stdout, stderr
+from sys import stderr, stdout
```
Comment (Kyle-B) [inline util/chplenv/chpl_aux_filesys.py]: No path change here?

Comment (ben-albrecht) [inline util/chplenv/chpl_aux_filesys.py]: See latest commit

File: util/chplenv/chpl_3p_re2_configs.py (hunk 1 above)
Comment (ben-albrecht) [inline util/chplenv/chpl_3p_re2_configs.py]: See latest commit.

**Comment by Kyle-B** [comment]
Looks fine to me.

**Pull request merged**